providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/grpc/GrpcStreamConnector.java

```diff
@@ -1,170 +0,0 @@
-    private final String providerId;
-    private final GrpcConnector<
-                    FlagSyncServiceGrpc.FlagSyncServiceStub, FlagSyncServiceGrpc.FlagSyncServiceBlockingStub>
-            grpcConnector;
-    private final LinkedBlockingQueue<GrpcResponseModel> streamReceiver;
-
-    /**
-     * Creates a new GrpcStreamConnector responsible for observing the event stream.
-     */
-    public GrpcStreamConnector(final FlagdOptions options, Consumer<FlagdProviderEvent> onConnectionEvent) {
-        deadline = options.getDeadline();
-        selector = options.getSelector();
-        providerId = options.getProviderId();
-        streamReceiver = new LinkedBlockingQueue<>(QUEUE_SIZE);
-        grpcConnector = new GrpcConnector<>(
-                options,
-                FlagSyncServiceGrpc::newStub,
-                FlagSyncServiceGrpc::newBlockingStub,
-                onConnectionEvent,
-                stub -> {
-                    String localSelector = selector;
-                    String localProviderId = providerId;
-
-                    final SyncFlagsRequest.Builder syncRequest = SyncFlagsRequest.newBuilder();
-                    if (localSelector != null) {
-                        syncRequest.setSelector(localSelector);
-                    }
-                    if (localProviderId != null) {
-                        syncRequest.setProviderId(localProviderId);
-                    }
-
-                    stub.syncFlags(syncRequest.build(), new GrpcStreamHandler(streamReceiver));
-                });
-    }
-
-    /** Initialize gRPC stream connector. */
-    public void init() throws Exception {
-        grpcConnector.initialize();
-        Thread listener = new Thread(() -> {
-            try {
-                observeEventStream(blockingQueue, shutdown, deadline);
-            } catch (InterruptedException e) {
-                log.warn("gRPC event stream interrupted, flag configurations are stale", e);
-                Thread.currentThread().interrupt();
-            }
-        });
-
-        listener.setDaemon(true);
-        listener.start();
-    }
-
-    /** Get blocking queue to obtain payloads exposed by this connector. */
-    public BlockingQueue<QueuePayload> getStream() {
-        return blockingQueue;
-    }
-
-    /**
-     * Shutdown gRPC stream connector.
-     *
-     * @throws InterruptedException if stream can't be closed within deadline.
-     */
-    public void shutdown() throws InterruptedException {
-        if (shutdown.getAndSet(true)) {
-            return;
-        }
-        this.grpcConnector.shutdown();
-    }
-
-    /** Contains blocking calls, to be used concurrently. */
-    void observeEventStream(final BlockingQueue<QueuePayload> writeTo, final AtomicBoolean shutdown, final int deadline)
-            throws InterruptedException {
-
-        log.info("Initializing sync stream observer");
-
-        while (!shutdown.get()) {
-            writeTo.clear();
```

Comment:
I've removed this queue clear. It actually caused a race condition because it's possible that the queue was cleared after a stream restart before it was consumed-from, leading to a lost event.